src/hooks/useSAMLSignIn.ts

```diff
@@ -12,9 +59,28 @@ export const useSAMLSignIn = (samlAuthProviderId?: string) => {
       samlAuthProviderId
     );
 
+    if (showLoginLoading) {
+      showLoginLoading();
+    }
+
     firebase
       .auth()
       .signInWithPopup(SAMLAuthProvider)
+      .then((result) => {
+        if (
+          !result.additionalUserInfo?.isNewUser ||
+          !result.user?.uid ||
+          !samlProfileMappings
+        )
+          return;
+
+        const mappedProfileData = getMappedValues(samlProfileMappings, result);
+        // TODO: Make a utils sessionStorage. Move PrefillProfileData into const/enum
+        sessionStorage.setItem(
```

Comment:
This is probably ok for now, but eventually this would probably be done some other way. Canonically I would say redux, but given we navigate with full page loads for now that won’t work. Probably good to add an @ debt comment here.

Reply:
We also have a file where we aim to centralise local storage/etc things. Like a utils file.

Reply:
> This is probably ok for now, but eventually this would probably be done some other way. Canonically I would say redux, but given we navigate with full page loads for now that won’t work. Probably good to add an @ debt comment here.

Yep, but not only this. I guess we would also want to handle a case, when something happens to a connection or a person closes the tab. We still want to have that data saved. My initial approach was to use URL params, but given the way the flow implemented now, it would take drastic changes to achieve it.

Reply:
The user will always auth via okta though, so we should be able to always read this same info off their login session I believe. Even directly in the create profile screen. So presumably we wouldn’t need to ‘cache and read back’ it at all?

Reply:
Changed in https://github.com/sparkletown/sparkle/pull/1667/commits/15dfaa4a6348a87a57db3386e982e12358de521c

Reply:
My original thoughts with using the profile was that we could just pass the URL directly into the profile image field within firestore rather than needing explicit logic in `src/components/molecules/ProfilePictureInput/ProfilePictureInput.tsx`/etc. Obviously if they chose to upload a new image then this method would no longer automagically work. Maybe it could just be an option shown on the 'default avatars' page when we have a `githubHandle` for them. Shown alongside the other 'defaults'. And pre-selected as the main avatar to use on initial profile creation.

Reply:
Actually.. disregard.. I think that may be more or less exactly what you did here.. `ProfilePictureInput` != `UserProfilePicture` as I originally read it as.